pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -221,11 +224,82 @@ func (builder *nodeEntityDTOBuilder) getNodeCommoditiesSold(node *api.Node, clus
 		Capacity(clusterCommodityDefaultCapacity).
 		Create()
 	if err != nil {
-		return nil, err
+		return nil, isAvailableForPlacement, err
 	}
 	commoditiesSold = append(commoditiesSold, clusterComm)
 
-	return commoditiesSold, nil
+	return commoditiesSold, isAvailableForPlacement, nil
+}
+
+func (builder *nodeEntityDTOBuilder) getNodeStorageCommoditiesSold(nodeName string) ([]*proto.CommodityDTO, bool) {
+	var resourceCommoditiesSold []*proto.CommodityDTO
+	vstorageResources := []string{"rootfs", "imagefs"}
+	entityType := metrics.NodeType
+	resourceType := metrics.VStorage
+	protoCommodityType := proto.CommodityDTO_VSTORAGE
+	isAvailableForPlacement := true
+
+	for _, resource := range vstorageResources {
+		entityID := nodeName
+		if resource == "imagefs" {
+			entityID = fmt.Sprintf("%s-%s", nodeName, resource)
```

Comment:
Will this be unique globally?

Reply:
This needs to be unique among the vstorage commodities on the give node. Is that not so?